libs/payments/ui/src/lib/client/components/CheckoutCheckbox.tsx

```diff
@@ -38,84 +39,87 @@ export function CheckoutCheckbox({
   };
 
   return (
-    <HoverCard.Root open={isRequired && !isChecked}>
-      <label className="flex gap-5 items-center mt-6">
-        <HoverCard.Trigger>
-          <input
-            type="checkbox"
-            name="confirm"
-            className="grow-0 shrink-0 basis-4 scale-150 cursor-pointer"
-            checked={isChecked}
-            onChange={changeHandler}
-          />
-        </HoverCard.Trigger>
-        {isClient && (
-          <Localized
-            id="next-payment-confirm-with-legal-links-static-3"
-            elems={{
-              termsOfServiceLink: (
-                <a
+    <Tooltip.Provider>
+      <Tooltip.Root open={isRequired && !isChecked}>
+        <label className="flex gap-5 items-center my-6">
+          <Tooltip.Trigger asChild>
+            <input
+              type="checkbox"
+              name="confirm"
+              className="grow-0 shrink-0 basis-4 scale-150 cursor-pointer"
+              checked={isChecked}
+              onChange={changeHandler}
+              required
+              aria-describedby="checkboxError"
+              aria-required
+            />
+          </Tooltip.Trigger>
+          {isClient && (
+            <Localized
+              id="next-payment-confirm-with-legal-links-static-3"
+              elems={{
+                termsOfServiceLink: (
+                  <LinkExternal
```

Comment:
I think all these 2-4 `<LinkExternal>` components will end up having the same default `data-testid` prop, if that's an issue. 🤷 
```js
'data-testid': testid = 'link-external',
```

Reply:
Updated, thanks Peter!